src/ShallowTraversal.js

```diff
@@ -29,7 +29,7 @@ export function childrenOfNode(node) {
 
 export function hasClassName(node, className) {
   const classes = propsOfNode(node).className || '';
-  return ` ${classes} `.indexOf(` ${className} `) > -1;
+  return ` ${classes.trim()} `.indexOf(` ${className} `) > -1;
```

Comment:
I guess I'm unsure of what the expected behavior is supposed to be in that case. This change was meant to just address non-significiant (leading/trailing) whitespace. If we're coercing all whitespace characters to spaces I think that's a bit different. A user might have some (strange) reason to have a new-tab there instead of a space?

Also:

```js
'foo\nbar'.replace(/\s*/g, ' ')
// " f o o  b a r "
```

Even if we just do `.replace(/\s/g, ' ')` cases like `foo\nbar\n` would return a string (`"foo bar "`) that still needs to be trimmed.